core/lib/dal/src/models/proto/mod.proto

```diff
@@ -23,3 +16,75 @@ message Payload {
   repeated Transaction transactions = 8;
   optional bool last_in_batch = 10; // required
 }
+
+message Transaction {
+  oneof common_data {
+    L1TxCommonData L1 = 1;
+    L2TxCommonData L2 = 2;
+    ProtocolUpgradeTxCommonData ProtocolUpgrade = 3;
```

Comment:
Bikeshedding: I think it's idiomatic to name `oneof` variants as fields (since on the wire level they are), i.e. in `snake_case`.

Reply:
Fixed.